### PR TITLE
add locale checker before appending locale to absolute url

### PR DESCRIPTION
**Pull request by OsmondJiang**
fix: https://ceapex.visualstudio.com/Engineering/_workitems/edit/98130/

### CODE REVIEW

File: src/docfx/localization/LocalizationUtility.cs [hunk 1]
```diff
@@ -2,18 +2,24 @@
...
 using System.IO;
+using System.Linq;
 using System.Text.RegularExpressions;
 
 namespace Microsoft.Docs.Build
 {
     internal static class LocalizationUtility
     {
+        private static readonly HashSet<string> s_locales = new HashSet<string>(CultureInfo.GetCultures(CultureTypes.AllCultures).Except(CultureInfo.GetCultures(CultureTypes.NeutralCultures)).Select(c => c.Name), StringComparer.OrdinalIgnoreCase);
         private static readonly Regex s_nameWithLocale = new Regex(@"^.+?(\.[a-z]{2,4}-[a-z]{2,4}(-[a-z]{2,4})?|\.loc)?$", RegexOptions.IgnoreCase);
```
Comment (yufeih) [inline src/docfx/localization/LocalizationUtility.cs]: can `s_nameWIthLocale` also share the same local list?

File: src/docfx/localization/LocalizationUtility.cs [hunk 2]
```diff
@@ -2,18 +2,24 @@
...
 using System.Globalization;
 using System.IO;
+using System.Linq;
 using System.Text.RegularExpressions;
 
 namespace Microsoft.Docs.Build
 {
     internal static class LocalizationUtility
     {
+        private static readonly HashSet<string> s_locales = new HashSet<string>(CultureInfo.GetCultures(CultureTypes.AllCultures).Except(CultureInfo.GetCultures(CultureTypes.NeutralCultures)).Select(c => c.Name), StringComparer.OrdinalIgnoreCase);
```
Comment (yufeih) [inline src/docfx/localization/LocalizationUtility.cs]: is `Contains('-')` a better condition to ensure we don't include strings without `-` into the locale list?

File: src/docfx/localization/LocalizationUtility.cs (hunk 1 above)
Comment (OsmondJiang) [reply src/docfx/localization/LocalizationUtility.cs]: yes we can, I will do it in another PR, it ends with this locale list

File: src/docfx/localization/LocalizationUtility.cs (hunk 2 above)
Comment (herohua) [reply src/docfx/localization/LocalizationUtility.cs]: I am wondering how is this list compared to the actual list Docs supports...

Comment (OsmondJiang) [reply src/docfx/localization/LocalizationUtility.cs]: I think it's the same from the summary:
```
  Cultures that are associated with a language but are not specific to a country/region.
        //     The names of .NET Framework cultures consist of the lowercase two-letter code
        //     derived from ISO 639-1. For example: &quot;en&quot; (English) is a neutral culture.
        NeutralCultures = 1,

Comment (OsmondJiang) [reply src/docfx/localization/LocalizationUtility.cs]: > I am wondering how is this list compared to the actual list Docs supports...

I added docs support locales to the test

**Comment by OsmondJiang** [comment]
expected diff:

image

File: src/docfx/localization/LocalizationUtility.cs (hunk 2 above)
Comment (yufeih) [reply src/docfx/localization/LocalizationUtility.cs]: I would like to suggest we start with the docs locale list

Comment (OsmondJiang) [reply src/docfx/localization/LocalizationUtility.cs]: don't want to add docs' limitation to docfx.

File: src/docfx/localization/LocalizationUtility.cs [hunk 3]
```diff
@@ -2,18 +2,24 @@
...
 using System.Globalization;
 using System.IO;
+using System.Linq;
 using System.Text.RegularExpressions;
 
 namespace Microsoft.Docs.Build
 {
     internal static class LocalizationUtility
     {
+        private static readonly HashSet<string> s_locales = new HashSet<string>(CultureInfo.GetCultures(CultureTypes.AllCultures).Except(CultureInfo.GetCultures(CultureTypes.NeutralCultures)).Select(c => c.Name).Concat(new[] { "zh-cn", "zh-tw", "zh-hk", "zh-sg", "zh-mo" }), StringComparer.OrdinalIgnoreCase);
```
Comment (live1206) [inline src/docfx/localization/LocalizationUtility.cs]: It would be helpful to have some comments why we concat some additional locales here

**Pull request merged**